src/buffer_reader.h

```diff
@@ -30,11 +30,11 @@ class LIBZIM_PRIVATE_API BufferReader : public Reader {
       : source(source) {}
     virtual ~BufferReader() {};
 
-    zsize_t size() const;
-    offset_t offset() const;
+    zsize_t size() const override;
+    offset_t offset() const override;
```

Comment:
This commit better be moved right after the one introducing `Reader::readImpl()`